terraform/apps/registry/bucket.tf

```diff
@@ -0,0 +1,88 @@
+resource "aws_s3_bucket" "sunodo_bucket" {
+  bucket        = var.bucket_name
+  force_destroy = false
+
+  tags = {
+    Name        = "sunodo_bucket"
+    Environment = "Dev"
+  }
+}
+
+resource "aws_s3_bucket_acl" "sunodo_bucket_acl" {
+  bucket = aws_s3_bucket.sunodo_bucket.id
+  acl    = "private"
+}
+
+resource "aws_s3_bucket_public_access_block" "sunodo_bucket_block_pub_access" {
+  bucket                  = aws_s3_bucket.sunodo_bucket.id
+  block_public_acls       = true
+  block_public_policy     = true
+  restrict_public_buckets = true
+}
+
+data "aws_iam_policy_document" "sunodo_registry" {
+  statement {
+
+    actions = [
+      "s3:PutObject",
+      "s3:GetObject",
+      "s3:DeleteObject",
+      "s3:ListMultipartUploadParts",
+      "s3:AbortMultipartUpload",
+      "s3:ListBucket",
+      "s3:GetBucketLocation",
+      "s3:ListBucketMultipartUploads",
+    ]
+    resources = [
+      aws_s3_bucket.sunodo_bucket.arn,
+      "${aws_s3_bucket.sunodo_bucket.arn}/*",
+    ]
+  }
+}
```

Comment:
This policy is different from the one defined at the suggested link

You see it has two statements, one for the `BUCKET_NAME`, and other for `BUCKET_NAME/*` (its contents).

https://github.com/docker/docs/blob/main/registry/storage-drivers/s3.md#s3-permission-scopes

Please make the proper adjustments.

Reply:
Fixed !